### PR TITLE
Add XCFun recipe

### PR DESCRIPTION
**Pull request by robertodr**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by robertodr** [comment]
I keep getting this error I don't understand on macOS:
```
  ERROR :: lib/libc++.1.dylib not in prefix_owners
  ERROR :: lib/libc++.1.dylib not in prefix_owners
   INFO (xcfun,lib/python3.7/site-packages/xcfun/_xcfun.cpython-37m-darwin.so): Needed DSO lib/libc++.1.dylib found in conda-forge::libcxx-9.0.1-1
   INFO (xcfun,lib/python3.7/site-packages/xcfun/_xcfun.cpython-37m-darwin.so): lib/libxcfun.2.dylib found in this package
   INFO (xcfun,lib/python3.7/site-packages/xcfun/_xcfun.cpython-37m-darwin.so): Needed DSO /usr/lib/libSystem.B.dylib found in the whitelist
   INFO (xcfun,lib/libxcfun.2.dylib): Needed DSO lib/libc++.1.dylib found in conda-forge::libcxx-9.0.1-1
   INFO (xcfun,lib/libxcfun.2.dylib): Needed DSO /usr/lib/libSystem.B.dylib found in the whitelist
```

File: recipes/xcfun/build.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+${BUILD_PREFIX}/bin/cmake \
+        -H${SRC_DIR} \
+        -Bbuild \
+        -GNinja \
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_CXX_COMPILER=${CXX} \
+        -DPYTHON_EXECUTABLE=${PYTHON} \
+        -DCMAKE_INSTALL_LIBDIR="lib" \
+        -DPYMOD_INSTALL_LIBDIR="/python${PY_VER}/site-packages" \
```
Comment (scopatz) [inline recipes/xcfun/build.sh]: I think you can use the `$SP_DIR` variable here

File: recipes/xcfun/build.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Mechanism to tweak compiler flags
+# Pass with -DCMAKE_<LANG>_FLAGS=${ALLOPTS}
+ALLOPTS="${CFLAGS}"
+
+# configure
+${BUILD_PREFIX}/bin/cmake \
```
Comment (scopatz) [inline recipes/xcfun/build.sh]: You can just call `cmake`

File: recipes/xcfun/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  license: MPL-2.0
+  license_url: https://opensource.org/licenses/MPL-2.0
+  license_file: LICENSE.md
+  license_family: MOZILLA
+  summary: "Exchange-Correlation functionals with arbitrary order derivatives."
+
+extra:
+  recipe-maintainers:
+    - robertodr
+    - bast
```
Comment (scopatz) [inline recipes/xcfun/meta.yaml]: @bast are you willing to be a maintainer?

Comment (bast) [reply recipes/xcfun/meta.yaml]: Yes

**Comment by robertodr** [comment]
Both the Linux and macOS CI lanes are finally green!

File: recipes/xcfun/bld.bat [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      -DCMAKE_INSTALL_LIBDIR="%LIBRARY_LIB%" ^
+      -DCMAKE_INSTALL_INCLUDEDIR="%LIBRARY_INC%" ^
+      -DCMAKE_INSTALL_BINDIR="%LIBRARY_BIN%" ^
+      -DCMAKE_INSTALL_DATADIR="%LIBRARY_PREFIX%" ^
+      -DPYMOD_INSTALL_LIBDIR="/../../Lib/site-packages" ^
+      -DCMAKE_CXX_FLAGS="/wd4018 /wd4101 /wd4996" ^
+      -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=true ^
+      -DXCFUN_PYTHON_INTERFACE=ON ^
+      -DXCFUN_MAX_ORDER=8 ^
+      -DPYTHON_EXECUTABLE="%PYTHON%"
```
Comment (scopatz) [inline recipes/xcfun/bld.bat:16]: This should be a release build

Comment (scopatz) [reply recipes/xcfun/bld.bat:16]: Oh it is done below nm

File: recipes/xcfun/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+requirements:
+  build:
+    - cmake >=3.11
+    - {{ compiler('cxx') }}
+    - pybind11 =*=*py{{ pyver }}*
+    - ninja
+  host:
+    - python
+  run:
+    - numpy
```
Comment (scopatz) [inline recipes/xcfun/meta.yaml:30]: Does this link against numpy or just use it at runtime?

Comment (robertodr) [reply recipes/xcfun/meta.yaml:30]: It just uses it at runtime

File: recipes/xcfun/bld.bat (hunk 4 above)
Comment (robertodr) [reply recipes/xcfun/bld.bat:16]: I should probably remove the `bld.bat` as the library doesn't currently build on windows.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**